GoogleCloudExtension/GoogleCloudExtensionUnitTests/StackdriverLogsViewer/LogsViewerToolWindowTests.cs

```
@@ -7,20 +11,94 @@ namespace GoogleCloudExtensionUnitTests.StackdriverLogsViewer
+        {
+            _objectUnderTest.Frame = _frameMock.Object;
+
+            Assert.IsNotNull(_objectUnderTest.ViewModel);
+        }
+
+        [TestMethod]
+        public void TestProjectIdChanged()
+        {
+            _objectUnderTest.Frame = _frameMock.Object;
+            ILogsViewerViewModel oldViewModel = _objectUnderTest.ViewModel;
+
+            CredentialsStore.Default.UpdateCurrentProject(new Project { ProjectId = "new-project-id" });
+
+            Assert.AreNotEqual(oldViewModel, _objectUnderTest.ViewModel);
+        }
+
+        [TestMethod]
+        public void TestProjectIdReset()
+        {
+            _objectUnderTest.Frame = _frameMock.Object;
+            ILogsViewerViewModel oldViewModel = _objectUnderTest.ViewModel;
+
+            CredentialsStore.Default.ResetCredentials(null, null);
+
+            Assert.AreNotEqual(oldViewModel, _objectUnderTest.ViewModel);
+        }
+
+        [TestMethod]
+        public void TestCloseDisablesProjectIdChanged()
+        {
+            _objectUnderTest.Frame = _frameMock.Object;
+            ILogsViewerViewModel oldViewModel = _objectUnderTest.ViewModel;
+
+            ((IVsWindowPane)_objectUnderTest).ClosePane();
+            CredentialsStore.Default.UpdateCurrentProject(new Project { ProjectId = "new-project-id" });
+
+            Assert.AreEqual(oldViewModel, _objectUnderTest.ViewModel);
+        }
+
+        [TestMethod]
+        public void TestCloseDisablesProjectIdReset()
+        {
+            _objectUnderTest.Frame = _frameMock.Object;
+            ILogsViewerViewModel oldViewModel = _objectUnderTest.ViewModel;
+
+            ((IVsWindowPane)_objectUnderTest).ClosePane();
+            CredentialsStore.Default.ResetCredentials(null, null);
+
+            Assert.AreEqual(oldViewModel, _objectUnderTest.ViewModel);
+        }
+
+        [TestMethod]
+        public void TestSetContentOfInvalidType()
+        {
+            _objectUnderTest.Content = new object();
```

Shouldn't this fail instead of silently skipping assignment?

---

Agreed. Done.